### PR TITLE
Upgrade ebpf-manager to v0.3.0

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Upgrade ebpf-manager to `v0.3.0`

### Motivation

Several performance improvements and bug fixes.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by paulcacheux — APPROVED** [review]

File: pkg/network/usm/ebpf_gotls.go [hunk 1]
```diff
@@ -233,6 +233,10 @@ func (p *GoTLSProgram) ConfigureOptions(options *manager.Options) {
 		options.MapEditors = make(map[string]*ebpf.Map)
 	}
 
+	options.MapSpecEditors[probes.SockByPidFDMap] = manager.MapSpecEditor{
+		MaxEntries: p.cfg.MaxTrackedConnections,
+		EditorFlag: manager.EditMaxEntries,
+	}
```
Comment (guyarb) [inline pkg/network/usm/ebpf_gotls.go:239]: why? we're getting the map FD and link it in the next line

Comment (brycekahle) [reply pkg/network/usm/ebpf_gotls.go:239]: We switched internally to using a non-deprecated way of doing the map editing, and it enforces that the key size, value size, map type, and *max entries* match before doing the replacement.

Comment (guyarb) [reply pkg/network/usm/ebpf_gotls.go:239]: and if it does not match, then we have a failure?

Comment (brycekahle) [reply pkg/network/usm/ebpf_gotls.go:239]: Yes, correct.

**Review by usamasaqib — APPROVED** [review]

**Review by guyarb — APPROVED** [review]

**Pull request merged**